ngi_visualizations/fpkm_scatter/fpkm_scatter.py

```diff
@@ -213,13 +263,100 @@ def plot_fpkm_scatter (sample_1, sample_2, x_lab, y_lab, output_fn=False, linear
     logging.info("Saving to {} and {}".format(png_fn, pdf_fn))
     plt.savefig(png_fn)
     plt.savefig(pdf_fn)
+    
+    # Return the filenames and R2
+    return {'png': png_fn, 'pdf': pdf_fn, output_fn:r_squared }
+
 
-    # Return the filenames
-    return {'png': png_fn, 'pdf': pdf_fn}
+def make_heatmap(data):
+    """
+    Takes a dict of r2 values and generates a heatmap using matplotlib
+    """
+    
+    #list the sample names
+    names=set()
+    for i in data.keys():
+        names.update(i.split('-'))
+    names=sorted(list(names))
+    
+    #If the sample looks like it's form a NGI project, do specific cleaning
+    ngi = re.compile(r'^P\d+_\d+')
+    clean_names=[]
+    for i in names:
+        i=i.split(".")[0]   #Remove everything after the first dot
+        j=re.search(ngi, i)
+        if j:
+            if i.endswith('Aligned'):
+                  i = i[:-7]
+        clean_names.append(i)
+    
+    matrix = []
+    #Generate the data matrix (list of lists) 
+    for idx, x in enumerate(names):
+        matrix.append([])
+        for idy, y in enumerate(names):
+        	v = 0
```

Comment:
You suddenly switch to using tabs here instead of spaces...

![tabs spaces](https://www.emacswiki.org/pics/static/TabsSpacesBoth.png)